docs/Architecture.md

```diff
@@ -1,2 +1,351 @@
 # Architecture
 
+## System Overview
+
+The Telegram Payments system enables secure, decentralized payments on Cosmos chains through Telegram. The architecture consists of three main components working together: a Telegram Bot for user interaction, a WAVS Operator Set for secure transaction validation, and Payment Rails smart contracts for on-chain execution.
+
+## High-Level Architecture Diagram
+
+```mermaid
+graph TB
+    subgraph "User Interface"
+        TU[Telegram User]
+        Wallet[User Wallet]
+        MiniApp[Telegram Mini-App]
+    end
+    
+    subgraph "Telegram Bot"
+        TBot[Backend Server]
+    end
+    
+    subgraph "WAVS Operator Network"
+        WAgg[WAVS Aggregator]
+        WO1[WAVS Operator 1]
+        WO2[WAVS Operator 2]
+        WO3[WAVS Operator 3]
+    end
+    
+    subgraph "Neutron Chain"
+        Bank[Bank Module]
+        PR[Payment Rails Contract]
+        SM[Service Manager Contract]
+    end
+    
+    TU -->|/status, /help, /start| TBot
+    TU -->|Opens| MiniApp
+    MiniApp -->|Connect| Wallet
+    Wallet -->|Payment Grant| Bank
+    Wallet -->|Register| PR
+    
+    Bank -->|Payment Events| TBot
+    TBot -->|Notifications| TU
+    
+    TU -->|/receive, /send| WO1
+    TU -->|/receive, /send| WO2
+    TU -->|/receive, /send| WO3
+    
+    WO1 -->|Verify & Sign| WAgg
+    WO2 -->|Verify & Sign| WAgg
+    WO3 -->|Verify & Sign| WAgg
+    
+    WAgg -->|Submit Transaction| PR
+    WAgg -->|Send confirmations| TU
+    
+    PR -->|Verify Signatures| SM
+    PR -->|Execute transfers| Bank
+```
+
+## Component Architecture
+
+### 1. Telegram Bot (Quick Operations)
```

Comment:
can we please clarify this section as the Bot server? i.e. there's the bot which Telegram operates, and then there's the server we operate, so for example users do not send commands directly to our server